### PR TITLE
Upstream changes from experimental-select-control back to @woocommerce/components

### PR DESCRIPTION
**Pull request by chihsuan**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/team-ghidorah/issues/148.

This PR applies the following changes from the WCCOM experimental select control fork.

1. Add a toggle button
2. Add the ability to customize menu item active style
3. Add `scrollIntoViewOnOpen` feature and `position` prop to the Menu component

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Run `pnpm --filter=@woocommerce/storybook run storybook`
2. Go to http://localhost:6007/?path=/story/woocommerce-admin-experimental-selectcontrol--toggle-button
3. Observe that a toggle button is displayed on the right of the input box.
4  Click on the toggle button
5. Observe that the menu is opened.
6. Make sure other story examples still work as expected

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Review by ilyasfoo — DISMISSED** [review]
Awesome work, @chihsuan. Tested well! 🚢

**Review by adrianduffell — DISMISSED** [review]
Tested well, thanks @chihsuan ! 🚀 

The control doesn't seem to be in place in the new product management screens so I couldn't test it there. cc @woocommerce/mothra in case you want to confirm this.

However it LGTM in an older iteration:

Screen Shot 2023-03-24 at 10 27 07 pm

**Comment by chihsuan** [comment]
Hey @adrianduffell or @ilyasfoo 👋  Could I get approval again? 🙏  I fixed a conflict, but it dismissed the reviews.

**Review by ilyasfoo — APPROVED** [review]

**Pull request merged**
